pyopencl/array.py

```diff
@@ -104,6 +104,24 @@ def _get_truedivide_dtype(obj1, obj2, queue):
     return result
 
 
+def _get_broadcasted_binary_op_result(obj1, obj2, cq, dtype_getter=None):
+    if dtype_getter is None:
+        dtype_getter = _get_common_dtype
+
+    if obj1.shape == obj2.shape:
+        return obj1._new_like_me(dtype_getter(obj1, obj2, cq),
+                                 cq)
+    elif obj1.shape == ():
+        return obj2._new_like_me(dtype_getter(obj1, obj2, cq),
+                                 cq)
+    elif obj2.shape == ():
+        return obj1._new_like_me(dtype_getter(obj1, obj2, cq),
```

Comment:
Doesn't this subsume the `assert`s below?

Reply:
Added it as there isn't quite a promise that control reaching those asserts would have reached "_get_broadcasted_binary_op_result".